src/plugins/vis_types/pie/public/pie_component.tsx

```diff
@@ -285,82 +284,107 @@ const PieComponent = (props: PieComponentProps) => {
     ? visData.columns[visParams.dimensions.splitRow[0].accessor]
     : undefined;
 
+  /**
+   * Checks whether data have all zero values.
+   * If so, the no data container is loaded.
+   */
+  const isAllZeros = useMemo(() => visData.rows.every((row) => row[metricColumn.id] === 0), [
+    visData.rows,
+    metricColumn,
+  ]);
+
+  /**
+   * Checks whether data have negative values.
+   * If so, the no data container is loaded.
+   */
+  const hasNegative = useMemo(
+    () =>
+      visData.rows.some((row) => {
+        const value = row[metricColumn.id];
+        return typeof value === 'number' && value < 0;
+      }),
+    [visData.rows, metricColumn]
+  );
+
   return (
     <div className="pieChart__container" data-test-subj="visTypePieChart">
-      <div className="pieChart__wrapper" ref={parentRef}>
-        <LegendToggle
-          onClick={toggleLegend}
-          showLegend={showLegend}
-          legendPosition={legendPosition}
-        />
-        <Chart size="100%">
-          <ChartSplit
-            splitColumnAccessor={splitChartColumnAccessor}
-            splitRowAccessor={splitChartRowAccessor}
-            splitDimension={splitChartDimension}
-          />
-          <Settings
-            debugState={window._echDebugStateFlag ?? false}
+      {(isAllZeros || hasNegative) && <VisualizationNoResults hasNegativeValues={hasNegative} />}
```

Comment:
nitpick: These two variables are always used combined, maybe we can store them as `canShowPieChart` and use it here as

```
{ !canShowPieChart ? <VisualizationNoResults hasNegativeValues={hasNegative} /> : ... }
```

Or more in general here a ternary operator can be used instead of boolean operators.

Reply:
Sure